src/components/DropOutBatches/DropOutBatchesProfile.js

```diff
@@ -0,0 +1,122 @@
+import { Button, Typography, Grid } from "@mui/material";
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { METHODS } from "../../services/api";
+import DropOut from "../BatchClassComponents/DropOut";
+
+function DropOutIndividualComponent(props) {
+  const { title, id, pathway_name, open, setOpen } = props;
+  const close = () => {
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <Grid container spacing={4}>
+        <Grid item>
+          <div>
+            <Typography variant="subtitle1" color="black">
+              {title}
+            </Typography>
+            <Button
+              variant="outlined"
+              color="primary"
+              style={{ margin: "10px 2px", borderRadius: 90, height: 30 }}
+            >
+              <Typography variant="body2">{pathway_name}</Typography>
+            </Button>
+          </div>
+        </Grid>
+        <Grid item>
+          <div
+            style={{
+              margin: "10px 20px",
+              cursor: "pointer",
+            }}
+            onClick={() => {
+              setOpen(true);
+            }}
+          >
+            <svg
+              width="14"
+              height="14"
+              viewBox="0 0 14 14"
+              fill="none"
+              xmlns="http://www.w3.org/2000/svg"
+            >
+              <path
+                d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z"
+                fill="#BDBDBD"
+              />
+            </svg>
+          </div>
+        </Grid>
+      </Grid>
+      <DropOut
+        open={open}
+        close={close}
+        title={title}
+        id={id}
+        unregister_all={true}
+      />
+    </>
+  );
+}
+
+function DropOutBatchesProfile() {
+  const [dropOutBatches, setDropOutBatches] = useState(null);
+  const [open, setOpen] = useState(false);
+
+  const user = useSelector(({ User }) => User);
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}users/EnrolledBatches`,
+      headers: {
+        accept: "application/json",
+        Authorization: user.data.token,
+      },
+    }).then((res) => {
+      if (res?.data?.length > 0) {
+        setDropOutBatches(res.data);
+      } else {
+        setDropOutBatches(null);
+      }
+    });
```

Comment:
the initial plan is to dont show anything if no batches and if we keep empty bracket we need to check length rather I used the null which is in falsy value

Reply:
this api is getting used only here in complete app . 
so is it needed?

Reply:
For your first point, it's fine to leave it as is.  To me, it was just more natural to check if the `length` of `dropOutBatches` is greater than `0` on line 100 instead rather than setting it to `null`, which suggests the data hasn't been loaded yet.

As far as Redux is concerned, my answer would be it depends.  Currently, we're not caching data from the back-end so if the user navigates away from the page and comes back, it'll make another request.  If we expect to always do this and we never expect to use the user's Enrolled Batches anywhere else, I'd say it's fine to leave as is. 

However, if we wanted to limit the back-end requests, we could store this data in our global state, so only when the user navigates away from merakilearn.org would it have to make a request for the user's Enrolled batches.  Of course, with the current setup, caching wouldn't work if say the user were to enroll in/drop out from batches separately on the Android app while staying on merakilearn.org.  In this case, the user's batches here would be incorrect.